functions/assert_metadata.pp

```diff
@@ -57,23 +61,26 @@ function simplib::assert_metadata (
     $metadata = load_module_metadata($module_name)
 
     if empty($metadata) {
-      fail("Could not find metadata for '${module_name}'")
+      fail("Could not find metadata for module '${module_name}'")
     }
 
     if $_options['os']['validate'] {
       if !$metadata['operatingsystem_support'] or empty($metadata['operatingsystem_support']) {
-        debug("'operatingsystem_support' was not found in '${module_name}'")
+        debug("'operatingsystem_support' was not found in module '${module_name}'")
+      }
+      elsif $facts['os']['name'] in $_options['blacklist'] {
+        fail("OS '${facts['os']['name']}' is not supported by module '${module_name}'")
```

Comment:
This "OS XXX is not supported by module YYY" message is misleading when the OS was blacklisted independently from the `metadata.json`; it should make the reason for the failure more specific in this case.

Reply:
@op-ct Can you think of a way to do this that's not going to require adding a ton of code everywhere?